docs/pages/upgrade-to-v3.md

```diff
@@ -0,0 +1,117 @@
+# Upgrade to XMTP V3
+
+The process to upgrade an app built with XMTP V2 to V3 is designed to be straightforward, with most functions in V3 working as they did in V2. However, there are some notable differences, which we cover here.
+
+:::info[Key takeaways]
+- Most core methods from V2 work similarly in V3, with some notable differences that are covered in this document.
+- We recommend that apps upgrade directly to V3, giving people access to a pure V3+ messaging experience with stronger encryption and laying the foundation for decentralization of the network. [Learn more](/upgrade-to-v3#faq)
+- To ensure continuity, Ephemera will provide a website where people can sign in with their V2 identities to access their V2 messages in a read-only format.
+
+:::
+
+## Core methods work similarly between V2 and V3
+
+Most core methods from V2, such as `newConversation`, `list`, `stream`, and `streamAllMessages`, work similarly in V3. 
+
+However, a key difference is that prior to V3, a conversation could represent a V1 or V2 conversation. In V3, a conversation can represent a group chat or direct message (DM).
+
+To learn more, see [Build a chat inbox](/inboxes/build-inbox).
+
+## Local databases mean no need to export/import topic data
+
+In V3, a client creates and manages a local database. For this reason, we no longer need to export or import topic data.
+
+## Loading messages
+
+- In V2, we used `listBatchMessages` to load all messages across conversations to help performantly load a conversation list in descending order by last message.
+- In V3, because there is a local database. we can simply use `list(order: .lastMessage)`.
+
+## Push notification differences
+
+### Decryption
+
+- In V1 and V2, we used intros V1 (`fromIntro`) and invites V2 (`fromInvite`) to decrypt invites and intros.
+- In V3, new conversations are called welcomes, and we use `fromWelcome` to decrypt them.
+
+### Getting topic IDs
+
+- In V2, we used `/xmtp/0/invite-$address/proto` to get the topic identifier for a new conversation.
+- In V3, we use `/xmtp/mls/1/w-$installationId/proto` to get the topic identifier for a new group chat or DM conversation.
+
+## Installation-specific private keys
+
```

Comment:
@nplasterer Did I capture what you shared about v2 vs v3 identity accurately here? <3